haystack/components/generators/chat/openai.py

```diff
@@ -358,9 +392,11 @@ async def run_async(
             tools_strict=tools_strict,
         )
 
-        chat_completion: Union[
-            AsyncStream[ChatCompletionChunk], ChatCompletion
-        ] = await self.async_client.chat.completions.create(**api_args)
+        openai_endpoint = api_args.pop("openai_endpoint")
+        if openai_endpoint == "parse":
+            chat_completion = await self.async_client.chat.completions.parse(**api_args)
+        else:
+            chat_completion = await self.async_client.chat.completions.create(**api_args)
```

Comment:
Let's use the same structure as in `run`
(`openai_endpoint_method = getattr(self.client.chat.completions, openai_endpoint)`, etc.)